plugins/inputs/webhooks/plex/plex_webhooks_models.go

```diff
@@ -0,0 +1,109 @@
+package plex
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+)
+
+const meas = "plex_webhooks"
+
+type Event interface {
+	NewMetric() telegraf.Metric
+}
+
+type PlexWebhookEvent struct {
+	Event    string   `json:"event"`
+	User     bool     `json:"user"`
+	Owner    bool     `json:"owner"`
+	Account  Account  `json:"Account"`
+	Server   Server   `json:"Server"`
+	Player   Player   `json:"Player"`
+	Metadata Metadata `json:"Metadata"`
+}
+
+type Account struct {
+	ID    int    `json:"id"`
+	Thumb string `json:"thumb"`
+	Title string `json:"title"`
+}
+
+type Server struct {
+	Title string `json:"title"`
+	UUID  string `json:"uuid"`
+}
+
+type Player struct {
+	Local         bool   `json:"local"`
+	PublicAddress string `json:"PublicAddress"`
+	Title         string `json:"title"`
+	UUID          string `json:"uuid"`
+}
+
+type Metadata struct {
+	LibrarySectionType   string `json:"librarySectionType"`
+	RatingKey            string `json:"ratingKey"`
+	Key                  string `json:"key"`
+	ParentRatingKey      string `json:"parentRatingKey"`
+	GrandparentRatingKey string `json:"grandparentRatingKey"`
+	GUID                 string `json:"guid"`
+	LibrarySectionID     int    `json:"librarySectionID"`
+	MediaType            string `json:"type"`
+	Title                string `json:"title"`
+	GrandparentKey       string `json:"grandparentKey"`
+	ParentKey            string `json:"parentKey"`
+	GrandparentTitle     string `json:"grandparentTitle"`
+	ParentTitle          string `json:"parentTitle"`
+	Summary              string `json:"summary"`
+	Index                int    `json:"index"`
+	ParentIndex          int    `json:"parentIndex"`
+	RatingCount          int    `json:"ratingCount"`
+	Thumb                string `json:"thumb"`
+	Art                  string `json:"art"`
+	ParentThumb          string `json:"parentThumb"`
+	GrandparentThumb     string `json:"grandparentThumb"`
+	GrandparentArt       string `json:"grandparentArt"`
+	AddedAt              int    `json:"addedAt"`
+	UpdatedAt            int    `json:"updatedAt"`
+}
+
+func (s PlexWebhookEvent) NewMetric() telegraf.Metric {
```

Comment:
I would make this a non-member function: `func NewMetric(e *PlexWebhookEvent) (telegraf.Metric, error)`